crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -166,29 +186,38 @@ impl Object {
             .extend()
     }
 
-    /// Objects can either be immutable, shared, owned by an address,
-    /// or are child objects (part of a dynamic field)
-    async fn kind(&self) -> Option<ObjectKind> {
+    /// The owner type of this Object.
+    /// Immutable and Shared Objects do not have owners.
+    async fn owner(&self, ctx: &Context<'_>) -> Option<ObjectOwnerNew> {
         use NativeOwner as O;
-        use ObjectKind as K;
-        Some(match self.native.owner {
-            O::AddressOwner(_) => K::Owned,
-            O::ObjectOwner(_) => K::Child,
-            O::Shared { .. } => K::Shared,
-            O::Immutable => K::Immutable,
-        })
-    }
 
-    /// The Address or Object that owns this Object.  Immutable and Shared Objects do not have
-    /// owners.
-    async fn owner(&self) -> Option<Owner> {
-        use NativeOwner as O;
-        let (O::AddressOwner(address) | O::ObjectOwner(address)) = self.native.owner else {
-            return None;
-        };
+        if self.native.owner.is_child_object() {
```

Comment:
Quite unsure if this is correct. I believe we want to know if this object is a child, and then make its owner to be a parent object.

Reply:
No i think this is right. is_child_object checks whether the owner of the object is an ObjectOwner, in which case the current object is a child object. You can alternatively check this behavior on the StoredObject itself, whether object.owner_type == 2 // for ObjectOwner